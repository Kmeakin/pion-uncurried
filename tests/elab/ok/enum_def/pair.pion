enum Pair(A: Type, B: Type) {
    MkPair(x: A, y: B),
}

let mkpair = MkPair;

let fst 
= fn(A: Type, B: Type, p: Pair(A, B)) => match p {
    MkPair(z, foo) => z, // Pair(A, B)
};


let fst_ann 
: fn (X: Type, Y: Type, q: Pair(X, Y)) -> X
= fn(A: Type, B: Type, p: Pair(A, B)) => match p {
    MkPair(z, foo) => z, // Pair(A, B)
};

