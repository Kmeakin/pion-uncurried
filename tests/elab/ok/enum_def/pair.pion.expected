enum Pair(A: Type, B: Type): Type {
    MkPair(x: A, y: B): Pair(A, B),
}

let mkpair: fn(A: Type, B: Type, x: A, y: B) -> Pair(A, B) = MkPair;

let fst: fn(A: Type, B: Type, p: Pair(A, B)) -> A = fn(A: Type, B: Type, p: Pair(A, B)) => match p {
    MkPair(z, foo) => z,
};

let fst_ann: fn(X: Type, Y: Type, q: Pair(X, Y)) -> X = fn(A: Type, B: Type, p: Pair(A, B)) => match p {
    MkPair(z, foo) => z,
};
